[sw,cryptolib] Change expected values for tests with result 0

h-filali opened the pull request:
This commit changes the expected values for the tests that have an expected result of 0. This makes sure we don't run into issues where the tests are not executing properly and we still get a correct result due to the expected value of 0.

Furthermore, this commit moves the tests to the hjson framework.

The input and output values don't change from before this change. This commit merely skips the unnecessary calculations.

Resolves #23343

andrea-caforio reviewed (approved):
Thanks @h-filali. No functional change just test logic.

h-filali commented:
I rebased and updated `sw/otbn/crypto/tests/p384_keygen_from_seed_test.s` to now check the unmasked secret key result. The masking seems to change slightly as randomness changes with different consumption patterns. To avoid having to update every time we make a change I now check the unmasked result.

andrea-caforio commented:
> I rebased and updated `sw/otbn/crypto/tests/p384_keygen_from_seed_test.s` to now check the unmasked secret key result. The masking seems to change slightly as randomness changes with different consumption patterns. To avoid having to update every time we make a change I now check the unmasked result.
> 
> Thanks @johannheyszl and @andrea-caforio for reviewing!

All good.
